databricks/koalas/indexes.py

```diff
@@ -110,15 +110,19 @@ def __init__(self, data: Union[DataFrame, list], dtype=None, name=None) -> None:
         if isinstance(data, DataFrame):
             assert dtype is None
             assert name is None
-            kdf = data
         else:
-            kdf = DataFrame(index=pd.Index(data=data, dtype=dtype, name=name))
-        internal = kdf._internal.copy(
-            spark_column=kdf._internal.index_spark_columns[0],
-            column_labels=kdf._internal.index_names,
+            data = DataFrame(index=pd.Index(data=data, dtype=dtype, name=name))
+
+        super(Index, self).__init__(data)
+
+    @property
+    def _internal(self) -> InternalFrame:
+        internal = self._kdf._internal
+        return internal.copy(
+            spark_column=internal.index_spark_columns[0],
```

Comment:
Should we maybe completely remove `spark_column` in `InternalFrame` since Series and Index don't hold them directly anymore. Maybe in a separate pr ..

Reply:
yes, I'm planning a clean-up later.